### ISSUE TITLE
Error when building by sbt outside of a git repository

### ISSUE THREAD

**Issue by Chuanqi1992** [allenai/science-parse#115]
There is an error when building a super-jar using sbt both in server and cli.

```
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
[info] Loading project definition from D:\science-parse-master\project
Missing bintray credentials C:\Users\v-chutan\.bintray\.credentials. Some bintray features depend on this.
Missing bintray credentials C:\Users\v-chutan\.bintray\.credentials. Some bintray features depend on this.
Missing bintray credentials C:\Users\v-chutan\.bintray\.credentials. Some bintray features depend on this.
Missing bintray credentials C:\Users\v-chutan\.bintray\.credentials. Some bintray features depend on this.
[info] Set current project to science-parse-master (in build file:/D:/science-parse-master/)
[warn] Credentials file C:\Users\v-chutan\.bintray\.credentials does not exist
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
[warn] Credentials file C:\Users\v-chutan\.bintray\.credentials does not exist
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
fatal: not a git repository (or any of the parent directories): .git
java.lang.RuntimeException: Nonzero exit code: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.Streamed$.sbt$Streamed$$next$1(ProcessImpl.scala:429)
        at sbt.Streamed$$anonfun$apply$10.apply(ProcessImpl.scala:432)
        at sbt.Streamed$$anonfun$apply$10.apply(ProcessImpl.scala:432)
        at sbt.AbstractProcessBuilder.lines(ProcessImpl.scala:151)
        at sbt.AbstractProcessBuilder.lines(ProcessImpl.scala:141)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$7.apply(VersionInjectorPlugin.scala:88)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$7.apply(VersionInjectorPlugin.scala:87)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
java.lang.RuntimeException: Nonzero exit value: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.AbstractProcessBuilder.getString(ProcessImpl.scala:134)
        at sbt.AbstractProcessBuilder.$bang$bang(ProcessImpl.scala:136)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$6.apply(VersionInjectorPlugin.scala:86)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$6.apply(VersionInjectorPlugin.scala:86)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
java.lang.RuntimeException: Nonzero exit value: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.AbstractProcessBuilder.getString(ProcessImpl.scala:134)
        at sbt.AbstractProcessBuilder.$bang$bang(ProcessImpl.scala:136)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply$mcJ$sp(VersionInjectorPlugin.scala:85)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply(VersionInjectorPlugin.scala:85)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply(VersionInjectorPlugin.scala:85)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
java.lang.RuntimeException: Nonzero exit code: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.Streamed$.sbt$Streamed$$next$1(ProcessImpl.scala:429)
        at sbt.Streamed$$anonfun$apply$10.apply(ProcessImpl.scala:432)
        at sbt.Streamed$$anonfun$apply$10.apply(ProcessImpl.scala:432)
        at sbt.AbstractProcessBuilder.lines(ProcessImpl.scala:151)
        at sbt.AbstractProcessBuilder.lines(ProcessImpl.scala:141)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$7.apply(VersionInjectorPlugin.scala:88)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$7.apply(VersionInjectorPlugin.scala:87)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
java.lang.RuntimeException: Nonzero exit value: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.AbstractProcessBuilder.getString(ProcessImpl.scala:134)
        at sbt.AbstractProcessBuilder.$bang$bang(ProcessImpl.scala:136)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$6.apply(VersionInjectorPlugin.scala:86)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$6.apply(VersionInjectorPlugin.scala:86)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
java.lang.RuntimeException: Nonzero exit value: 128
        at scala.sys.package$.error(package.scala:27)
        at sbt.AbstractProcessBuilder.getString(ProcessImpl.scala:134)
        at sbt.AbstractProcessBuilder.$bang$bang(ProcessImpl.scala:136)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply$mcJ$sp(VersionInjectorPlugin.scala:85)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply(VersionInjectorPlugin.scala:85)
        at org.allenai.plugins.VersionInjectorPlugin$$anonfun$1.apply(VersionInjectorPlugin.scala:85)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$3$$anonfun$apply$2.apply(System.scala:44)
        at sbt.std.Transform$$anon$4.work(System.scala:63)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1$$anonfun$apply$1.apply(Execute.scala:228)
        at sbt.ErrorHandling$.wideConvert(ErrorHandling.scala:17)
        at sbt.Execute.work(Execute.scala:237)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.Execute$$anonfun$submit$1.apply(Execute.scala:228)
        at sbt.ConcurrentRestrictions$$anon$4$$anonfun$1.apply(ConcurrentRestrictions.scala:159)
        at sbt.CompletionService$$anon$2.call(CompletionService.scala:28)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)
[error] (cli/*:gitRemotes) Nonzero exit code: 128
[error] (cli/*:gitSha1) Nonzero exit value: 128
[error] (cli/*:gitCommitDate) Nonzero exit value: 128
[error] (core/*:gitRemotes) Nonzero exit code: 128
[error] (core/*:gitSha1) Nonzero exit value: 128
[error] (core/*:gitCommitDate) Nonzero exit value: 128
[error] Total time: 1 s, completed Aug 3, 2018 8:18:55 AM
```

**Comment by dirkgr** [comment]
Unfortunately, you can only build inside of a git repository right now. This is a limitation of one of the SBT plugins we use.

So the correct way of building this is to do
 * `git clone https://github.com/allenai/science-parse.git`
 * `cd science-parse`
 * `sbt server/assembly` (or some other project, if you don't want to build the server)

**Comment by dirkgr** [comment]
I just released version 2.0.3, which hopefully fixes this problem. Cheers!
